src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -414,62 +415,90 @@ internal static bool IsModuleMatchingModuleSpec(PSModuleInfo moduleInfo, ModuleS
-            if (manifestPath != null &&
-                manifestPath.EndsWith(StringLiterals.PowerShellDataFileExtension, StringComparison.OrdinalIgnoreCase))
+            try
             {
-                try
-                {
-                    var dataFileSetting =
-                        PsUtils.GetModuleManifestProperties(
-                            manifestPath,
-                            PsUtils.ManifestModuleVersionPropertyName);
+                Hashtable dataFileSetting =
+                    PsUtils.GetModuleManifestProperties(
+                        manifestPath,
+                        PsUtils.ManifestModuleVersionPropertyName);
 
-                    var versionValue = dataFileSetting["ModuleVersion"];
-                    if (versionValue != null)
+                object versionValue = dataFileSetting["ModuleVersion"];
+                if (versionValue != null)
+                {
+                    Version moduleVersion;
+                    if (LanguagePrimitives.TryConvertTo(versionValue, out moduleVersion))
                     {
-                        Version moduleVersion;
-                        if (LanguagePrimitives.TryConvertTo(versionValue, out moduleVersion))
-                        {
-                            return moduleVersion;
-                        }
+                        return moduleVersion;
                     }
                 }
-                catch (PSInvalidOperationException)
-                {
-                }
             }
+            catch (PSInvalidOperationException) { }
 
             return new Version(0, 0);
         }
 
         internal static Guid GetManifestGuid(string manifestPath)
         {
-            if (manifestPath != null &&
-                manifestPath.EndsWith(StringLiterals.PowerShellDataFileExtension, StringComparison.OrdinalIgnoreCase))
+            try
             {
-                try
+                Hashtable dataFileSetting =
+                    PsUtils.GetModuleManifestProperties(
+                        manifestPath,
+                        PsUtils.ManifestGuidPropertyName);
+
+                object guidValue = dataFileSetting["GUID"];
+                if (guidValue != null)
                 {
-                    var dataFileSetting =
-                        PsUtils.GetModuleManifestProperties(
-                            manifestPath,
-                            PsUtils.ManifestGuidPropertyName);
+                    Guid guidID;
+                    if (LanguagePrimitives.TryConvertTo(guidValue, out guidID))
+                    {
+                        return guidID;
+                    }
+                }
+            }
+            catch (PSInvalidOperationException) { }
 
-                    var guidValue = dataFileSetting["GUID"];
-                    if (guidValue != null)
+            return new Guid();
+        }
+
+        internal static ExperimentalFeature[] GetExperimentalFeature(string manifestPath)
```

Comment:
What are the reasons for using an array here over `[I]List<ExperimentalFeature>` or `[I]Collection<ExperimentalFeature>`, or even a read-only collection?

I'm guessing we may still want to manipulate the datastructure somewhat (in an efficient way) without adding elements to it, but just want to ask the question anyway

Reply:
I think I use array just for using the shared empty array in case no experimental feature is defined.
Sure I can change the return type to `IList<>` or `ICollection<>` but I don't see much value of that change.